clippy_lints/src/duplicate_match_guards.rs

```diff
@@ -0,0 +1,171 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::{
+    HasSession, SpanRangeExt, indent_of, reindent_multiline, snippet_with_applicability, trim_span,
+};
+use clippy_utils::{eq_expr_value, span_contains_comment};
+use rustc_errors::Applicability;
+use rustc_hir::{Arm, ExprKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::declare_lint_pass;
+use rustc_span::BytePos;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for the same condition being checked in a match guard and in the match body
+    ///
+    /// ### Why is this bad?
+    /// This is usually just a typo or a copy and paste error.
+    ///
+    /// ### Known problems
+    /// False negatives: if the condition is an impure function, it could've been called twice on
+    /// purpose for its side effects
```

Comment:
I'm slightly confused by this section. This says "false negative", but then describes what sounds like would be a false *positive* ("the lint falsely warns if the condition has impure function calls which would change semantics"), but this case is actually handled by the lint anyway given that there's a test for it (so there's no issue at all)?